### PR TITLE
fix: handle empty predictions in batch formatting and add format_prediction (#1101)

### LINKED ISSUE #1101: Add empty box parser inside utilities.format_geometry
When an image has no predictions, it would be nice if utilities.format_geometry natively processed this into the output. Right now we have to do something like

https://github.com/weecology/DeepForest/blob/d0d6c0a7310dad417d0454ec3e36acf81dfe7340/src/deepforest/main.py#L908

```
predictions = m.predict_step(images, batch_index)
for pred in predictions:
    geometry = determine_geometry(pred)
    if len(pred[geometry]) == 0:
            y_pred = {}
            y_pred["y"] = torch.zeros(4)
            y_pred["labels"] = torch.zeros(1)
            y_pred["scores"] = torch.zeros(1)
    else:
            format_geometry(pred, geom_type=geometry)

```

# Desired functionality

1. a new single function 'format_prediction' that looks at a prediction dictionary, uses determine_geometry, format_geometry, and fills in the edge case of empty images.
2. Minimal tests that shows this behavior for points and boxes, @jveitchmichaelis we don't have polygon prediction yet, right?

### PR DESCRIPTION
**Pull request by hardikkaurani**
## Summary

Fixes #1101.

When an image has zero detections, prediction formatting previously required ad-hoc handling for empty prediction dictionaries. `predict_batch` also assumed box predictions and could drop empty results, causing failures or loss of image-to-result alignment for non-box models.

This PR introduces `utilities.format_prediction()` as a unified empty-safe formatting entry point while preserving the existing `format_geometry()` behavior for existing consumers.

### Changes

1. **Add `utilities.format_prediction()`**
   - Handles empty prediction dictionaries without manufacturing detections.
   - Returns a genuine 0-row `pandas.DataFrame` with the appropriate schema.
   - Supports box, point, and polygon prediction types.
   - Delegates non-empty predictions to `format_geometry()`.

2. **Harden `determine_geometry_type()`**
   - Safely handles empty GeoDataFrames.
   - Avoids `IndexError` when no geometry rows are available.

3. **Update `predict_batch()`**
   - Uses the model task to format predictions.
   - Supports box, point, and polygon predictions.
   - Preserves 1:1 image-to-result alignment, including images with zero detections.

4. **Preserve existing behavior**
   - `format_geometry(empty)` continues to return `None`.
   - Existing validation/callback consumers remain compatible with that contract.

### Difference from previous PR #1318

A previous unmerged PR (#1318) also explored adding `format_prediction()` for issue #1101.

This implementation differs in several important ways:

- Empty predictions are represented as genuine 0-row DataFrames rather than synthetic zero-coordinate detections.
- Geometry-specific schemas are preserved for box, point, and polygon predictions.
- `predict_batch()` no longer assumes every model produces `"boxes"` and preserves batch alignment when predictions are empty.
- Empty GeoDataFrame handling is hardened.
- The change is based on the current `main` branch and is limited to four focused files.

### Testing

- `pytest tests/test_utilities.py tests/test_main.py` — 143 passed
- `ruff check src/deepforest/utilities.py src/deepforest/main.py tests/test_utilities.py tests/test_main.py` — passed
- `git diff --check` — passed

No dependency or lockfile changes are included.

### CODE REVIEW

File: src/deepforest/main.py [hunk 1]
```diff
@@ -1114,13 +1114,10 @@ def predict_batch(self, images, preprocess_fn=None):
...
-        for pred in predictions:
-            if len(pred["boxes"]) == 0:
-                continue
-            geom_type = utilities.determine_geometry_type(pred)
-            result = utilities.format_geometry(pred, geom_type=geom_type)
-            results.append(result)
+        results = [
+            utilities.format_prediction(pred, scores=True, geom_type=self.model.task)
+            for pred in predictions
+        ]
```
[the remark this answers is not in the document]
Comment (hardikkaurani) [reply src/deepforest/main.py:1121]: Resolved in commit 37fc62c by using getattr(self.model, \task\, None) with a fallback to prediction key inference, and added regression test 	est_predict_batch_custom_model_without_task.
